deepmd/pd/model/descriptor/dpa1.py

```
@@ -0,0 +1,689 @@
+        h2
+            The rotationally equivariant pair-partical representation.
+            shape: nf x nloc x nnei x 3
+        sw
+            The smooth switch function. shape: nf x nloc x nnei
+
+        """
+        # cast the input to internal precsion
+        extended_coord = extended_coord.to(dtype=self.prec)
+        del mapping
+        nframes, nloc, nnei = nlist.shape
+        nall = extended_coord.reshape([nframes, -1]).shape[1] // 3
```

Ruff (0.8.0)

Local variable `nall` is assigned to but never used

Remove assignment to unused variable `nall`

(F841)